File: 0090-subsets-ii/0090-subsets-ii.py
def subsets(ind,n,arr,temp,res):
    if(ind>=n):
        temp2 =temp[:]
        temp2.sort()
        if(temp2 not in res):
            res.append(temp2)
        return
    
    temp.append(arr[ind])
    subsets(ind+1,n,arr,temp,res)

    temp.pop()
    subsets(ind+1,n,arr,temp,res)


class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        temp = []
        res = []
        subsets(0,n,nums,temp,res)
        return(res)

File: 0090-subsets-ii/0090-subsets-ii.cpp
void seqn(vector<int>& nums,int n,vector<vector<int>>& res,vector<int>& arr,int index){
    res.push_back(arr);
    for(int i=index;i<n;i++){
        if(i!=index && nums[i]==nums[i-1]){
            continue;
        }
        arr.push_back(nums[i]);
        seqn(nums,n,res,arr,i+1);
        arr.pop_back();
    }
}

class Solution {
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> arr;
        sort(nums.begin(),nums.end());
        seqn(nums,nums.size(),res,arr,0);
        return res;
    }
};